types/key/nl.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package key
+
+import (
+	"crypto/ed25519"
+
+	"go4.org/mem"
+	"tailscale.com/tka"
+	"tailscale.com/types/structs"
+)
+
+const (
+	// nlPrivateHexPrefix is the prefix used to identify a
+	// hex-encoded network-lock key.
+	nlPrivateHexPrefix = "nlpriv:"
+)
+
+// NLPrivate is a node-managed network-lock key, used for signing
+// node-key signatures and authority update messages.
+type NLPrivate struct {
+	_ structs.Incomparable // because == isn't constant-time
+	k [ed25519.PrivateKeySize]byte
```

Comment:
any particular reason to use a byte array rather than ed25519.PrivateKey?

Reply:
Not really, I'm mostly following the pattern of using an array rather than a slice, I presume this established as the pattern to minimize allocations.